docs/source/using/image-caching.adoc

```diff
@@ -4,39 +4,121 @@ include::variables.adoc[]
+
+To view currently cached images on the host:
 
 ----
-$ minishift config unset image-caching
+$ minishift image list
+openshift/origin-docker-registry:v3.6.0
+openshift/origin-haproxy-router:v3.6.0
+openshift/origin:v3.6.0
+----
+
+To view images available in the Docker daemon:
+
+----
+$ minishift image list --vm
+openshift/origin-deployer:v3.6.0
+openshift/origin-docker-registry:v3.6.0
+openshift/origin-haproxy-router:v3.6.0
+openshift/origin-pod:v3.6.0
+openshift/origin:v3.6.0
+----
+
+[[persisting-image-names]]
+=== Persisting Cached Image Names
+
+In order to avoid having to type the image names explicitly as part of the `image export` and `image import` command, you can persist the default image names for import and export in the persistent configuration:
+
+----
+$ minishift config set cache-images <image-name-0>,<image-name-1>
 ----
 
+Once the image names are stored in the persistent configuration, you can run `minishift image export` and `minishift image import` without any arguments.
+
+[[all-images]]
+=== Exporting and Importing All Images
+
+Last but not least, you can export and import all images using the `--all` flag.
+For the export command this means that all images currently available on the Docker daemon will be exported to the host.
+For the import command it means that all images available in the local {project} cache will be imported into the Docker daemon of the {project} VM.
+
+[WARNING]
+====
+Exporting and importing all images can take a long time and locally cached images can take up a considerable amount of disk space. We recommend to use this feature cautiously.
+====
+
+[[implicit-image-caching]]
+== Implicit image caching
+
+Image caching can be configured to automatically occur during the creation of the {project} VM while executing `minishift start`.
+To enable this feature you need to enable the _image-caching_ property in the persistent configuration using the xref:../command-ref/minishift_config_set#[`minishift config set`] command:
+
+----
+$ minishift config set image-caching true
+----
+
+Once enabled, caching occurs transparently, in a background process, the first time you use the xref:../command-ref/minishift_start#[`minishift start`] command.
+Once the images are cached under _$MINISHIFT_HOME/cache/images_, successive {project} VM creations will use these cached images.
+
 [NOTE]
 ====
-Image caching is considered experimental and its semantics and API are subject to change.
-The aim is to allow caching of arbitrary images, as well as using a better format for storing the images on the host.
-You can track the progress on this feature on GitHub issue link:https://github.com/minishift/minishift/issues/952[#952].
+Enabling implicit image caching will transparently add the required OpenShift images to the list of cached images as specified per `cache-images` configuration option. See xref:../using/image-caching.adoc#persisting-image-names[Persisting Cached Image Names].
```

Comment:
I think, it would be good to mention here the three images that are being cached as part of implicit caching.